File: Modelo/Producto.cs
namespace punto_de_venta_C_.Modelo
{
    public class Producto
    {
        private string id;
        private string nombre;
        private double precio;
        private double stock;
        private double stockMinimo;
        private string categoria;
        private string unidad;
        private string imagenRuta;

        public Producto(string id, string nombre, double precio, double stock, string categoria, string unidad, string imagenRuta)
        {
            this.id = id;
            this.nombre = nombre;
            this.precio = precio;
            this.stock = stock;
            this.stockMinimo = 5;
            this.categoria = categoria;
            this.unidad = unidad;
            this.imagenRuta = imagenRuta;
        }

        public string getId() { return id; }
        public void setId(string id) { this.id = id; }
        public string getNombre() { return nombre; }
        public void setNombre(string nombre) { this.nombre = nombre; }
        public double getPrecio() { return precio; }
        public void setPrecio(double precio) { this.precio = precio; }
        public double getStock() { return stock; }
        public void setStock(double stock) { this.stock = stock; }
        public double getStockMinimo() { return stockMinimo; }
        public void setStockMinimo(double min) { this.stockMinimo = min; }
        public string getCategoria() { return categoria; }
        public void setCategoria(string categoria) { this.categoria = categoria; }
        public string getUnidad() { return unidad; }
        public void setUnidad(string unidad) { this.unidad = unidad; }
        public string getImagenRuta()
        {
            if (string.IsNullOrWhiteSpace(imagenRuta))
            {
                try
                {
                    int numericId = int.Parse(id);
                    return "Images/" + numericId + ".jpg";
                }
                catch
                {
                    return "Images/" + id + ".jpg";
                }
            }
            return imagenRuta;
        }
        public void setImagenRuta(string imagenRuta) { this.imagenRuta = imagenRuta; }

        public bool esPorPieza() { return "Piezas".Equals(unidad); }
        public bool stockBajo() { return stock <= stockMinimo; }
        public void reducirStock(double cantidad)
        {
            this.stock = System.Math.Max(0, this.stock - cantidad);
            this.stock = System.Math.Round(this.stock * 1000.0) / 1000.0;
        }
        public void aumentarStock(double cantidad)
        {
            this.stock += cantidad;
            this.stock = System.Math.Round(this.stock * 1000.0) / 1000.0;
        }

        public override string ToString()
        {
            return nombre + " - $" + precio.ToString("0.00") + " [Stock: " + stock + "]";
        }
    }
}

File: Modelo/Producto.py
class Producto:
    def __init__(self, id='', nombre='', precio=0.0, stock=0.0, categoria='', unidad='', imagenRuta=''):
        self._id = id
        self._nombre = nombre
        self._precio = precio
        self._stock = stock
        self._stockMinimo = 5.0
        self._categoria = categoria
        self._unidad = unidad
        self._imagenRuta = imagenRuta

    def getId(self):
        return self._id

    def setId(self, id):
        self._id = id

    def getNombre(self):
        return self._nombre

    def setNombre(self, nombre):
        self._nombre = nombre

    def getPrecio(self):
        return self._precio

    def setPrecio(self, precio):
        self._precio = precio

    def getStock(self):
        return self._stock

    def setStock(self, stock):
        self._stock = stock

    def getStockMinimo(self):
        return self._stockMinimo

    def setStockMinimo(self, stockMinimo):
        self._stockMinimo = stockMinimo

    def getCategoria(self):
        return self._categoria

    def setCategoria(self, categoria):
        self._categoria = categoria

    def getUnidad(self):
        return self._unidad

    def setUnidad(self, unidad):
        self._unidad = unidad

    def getImagenRuta(self):
        if not self._imagenRuta or not self._imagenRuta.strip():
            try:
                numericId = int(self._id)
                return f"Images/{numericId}.jpg"
            except Exception:
                return f"Images/{self._id}.jpg"
        return self._imagenRuta

    def setImagenRuta(self, imagenRuta):
        self._imagenRuta = imagenRuta

    def esPorPieza(self):
        return self._unidad == "Piezas"

    def stockBajo(self):
        return self._stock <= self._stockMinimo

    def reducirStock(self, cantidad):
        self._stock = max(0.0, self._stock - cantidad)
        self._stock = round(self._stock, 3)

    def aumentarStock(self, cantidad):
        self._stock += cantidad
        self._stock = round(self._stock, 3)

    def __str__(self):
        return f"{self._nombre} - ${self._precio:.2f} [Stock: {self._stock}]"